test/core/httpcli/httpcli_test.c

```diff
@@ -47,56 +51,93 @@ static gpr_timespec n_seconds_time(int seconds) {
 }
 
 static void on_finish(void *arg, const grpc_httpcli_response *response) {
+  const char *expect = 
+      "<html><head><title>Hello world!</title></head>"
+      "<body><p>This is a test</p></body></html>";
   GPR_ASSERT(arg == (void *)42);
   GPR_ASSERT(response);
   GPR_ASSERT(response->status == 200);
+  GPR_ASSERT(response->body_length == strlen(expect));
+  GPR_ASSERT(0 == memcmp(expect, response->body, response->body_length));
   gpr_event_set(&g_done, (void *)1);
 }
 
-static void test_get(int use_ssl) {
+static void test_get(int use_ssl, int port) {
   grpc_httpcli_request req;
+  char* host;
 
   gpr_log(GPR_INFO, "running %s with use_ssl=%d.", "test_get", use_ssl);
 
+  gpr_asprintf(&host, "localhost:%d", port);
+  gpr_log(GPR_INFO, "requesting from %s", host);
+
   gpr_event_init(&g_done);
   memset(&req, 0, sizeof(req));
-  req.host = "www.google.com";
-  req.path = "/";
+  req.host = host;
+  req.path = "/get";
   req.use_ssl = use_ssl;
 
   grpc_httpcli_get(&req, n_seconds_time(15), on_finish, (void *)42);
+  gpr_free(host);
   GPR_ASSERT(gpr_event_wait(&g_done, n_seconds_time(20)));
 }
 
-/*
-static void test_post(int use_ssl) {
+static void test_post(int use_ssl, int port) {
   grpc_httpcli_request req;
+  char* host;
 
   gpr_log(GPR_INFO, "running %s with use_ssl=%d.", "test_post", (int)use_ssl);
 
+  gpr_asprintf(&host, "localhost:%d", port);
+  gpr_log(GPR_INFO, "posting to %s", host);
+
   gpr_event_init(&g_done);
   memset(&req, 0, sizeof(req));
-  req.host = "requestb.in";
-  req.path = "/1eamwr21";
+  req.host = host;
+  req.path = "/post";
   req.use_ssl = use_ssl;
 
-  grpc_httpcli_post(&req, NULL, 0, n_seconds_time(15), on_finish,
+  grpc_httpcli_post(&req, "hello", 5, n_seconds_time(15), on_finish,
                     (void *)42);
   GPR_ASSERT(gpr_event_wait(&g_done, n_seconds_time(20)));
 }
-*/
 
 int main(int argc, char **argv) {
+  gpr_subprocess* server;
+  char *me = argv[0];
+  char *lslash = strrchr(me, '/');
+  char* args[4];
+  char root[1024];
+  int port = grpc_pick_unused_port_or_die();
+
+  /* figure out where we are */
+  if (lslash) {
+    memcpy(root, me, lslash - me);
+    root[lslash - me] = 0;
+  } else {
+    strcpy(root, ".");
+  }
+
+  /* start the server */
+  gpr_asprintf(&args[0], "%s/../../test/core/httpcli/test_server.py", root);
+  args[1] = "--port";
+  gpr_asprintf(&args[2], "%d", port);
+  server = gpr_subprocess_create(3, (const char**)args);
```

Comment:
That made me smile btw. It might not work under Windows properly, but we'll see.